seatunnel-connectors-v2/connector-clickhouse/src/main/java/org/apache/seatunnel/connectors/seatunnel/clickhouse/sink/client/ClickhouseSinkFactory.java

```diff
@@ -0,0 +1,158 @@
+import static org.apache.seatunnel.connectors.seatunnel.clickhouse.config.ClickhouseConfig.SHARDING_KEY;
+import static org.apache.seatunnel.connectors.seatunnel.clickhouse.config.ClickhouseConfig.SPLIT_MODE;
+import static org.apache.seatunnel.connectors.seatunnel.clickhouse.config.ClickhouseConfig.SUPPORT_UPSERT;
+import static org.apache.seatunnel.connectors.seatunnel.clickhouse.config.ClickhouseConfig.TABLE;
+import static org.apache.seatunnel.connectors.seatunnel.clickhouse.config.ClickhouseConfig.USERNAME;
+
+@AutoService(Factory.class)
+public class ClickhouseSinkFactory implements TableSinkFactory {
+
+    @Override
+    public TableSink<SeaTunnelRow, ClickhouseSinkState, CKCommitInfo, CKAggCommitInfo> createSink(
+            TableSinkFactoryContext context) {
+        ReadonlyConfig readonlyConfig = context.getOptions();
+        CatalogTable catalogTable = context.getCatalogTable();
+        List<ClickHouseNode> nodes = ClickhouseUtil.createNodes(readonlyConfig);
+        Properties clickhouseProperties = new Properties();
+        readonlyConfig
+                .get(CLICKHOUSE_CONFIG)
+                .forEach((key, value) -> clickhouseProperties.put(key, String.valueOf(value)));
+
+        clickhouseProperties.put("user", readonlyConfig.get(USERNAME));
+        clickhouseProperties.put("password", readonlyConfig.get(PASSWORD));
+
+        ClickhouseProxy proxy = new ClickhouseProxy(nodes.get(0));
+        Map<String, String> tableSchema = proxy.getClickhouseTableSchema(readonlyConfig.get(TABLE));
+        String shardKey = null;
+        String shardKeyType = null;
+        ClickhouseTable table =
+                proxy.getClickhouseTable(readonlyConfig.get(DATABASE), readonlyConfig.get(TABLE));
+        if (readonlyConfig.get(SPLIT_MODE)) {
+            if (!"Distributed".equals(table.getEngine())) {
+                throw new ClickhouseConnectorException(
+                        CommonErrorCodeDeprecated.ILLEGAL_ARGUMENT,
+                        "split mode only support table which engine is "
+                                + "'Distributed' engine at now");
+            }
+            if (readonlyConfig.getOptional(SHARDING_KEY).isPresent()) {
+                shardKey = readonlyConfig.get(SHARDING_KEY);
+                shardKeyType = tableSchema.get(shardKey);
+            }
+        }
+        ShardMetadata metadata =
+                new ShardMetadata(
+                        shardKey,
+                        shardKeyType,
+                        table.getSortingKey(),
+                        readonlyConfig.get(DATABASE),
+                        readonlyConfig.get(TABLE),
+                        table.getEngine(),
+                        readonlyConfig.get(SPLIT_MODE),
+                        new Shard(1, 1, nodes.get(0)),
+                        readonlyConfig.get(USERNAME),
+                        readonlyConfig.get(PASSWORD));
+        proxy.close();
```

Comment:
add `try.. catch and finally` to close `proxy`?

Reply:
done